google/cloud/storage/oauth2/google_credentials.cc

```diff
@@ -32,9 +32,21 @@ namespace storage {
 inline namespace STORAGE_CLIENT_NS {
 namespace oauth2 {
 
-/// Parses the JSON file at `path` and creates the appropriate Credentials type.
+constexpr char kAdcLink[] =
+    "https://developers.google.com/identity/protocols/"
+    "application-default-credentials";
+
+// Parses the JSON or P12 file at `path` and creates the appropriate
+// Credentials type.
+//
+// If `service_account_scopes` or `service_account_subject` are specified, the
+// file at `path` must be a P12 service account or a JSON service account. If
+// a different type of credential file is found, this function returns
+// nullptr to indicate a service account file wasn't found.
 StatusOr<std::unique_ptr<Credentials>> LoadCredsFromPath(
-    std::string const& path) {
+    std::string const& path, bool non_service_account_ok,
```

Comment:
No change necessary, but I think we should be careful about using bool parameters, *unless* they are the only argument and their meaning is obvious from the function's name. The reason is that they usually lead to unclear call sites because the sense of their meaning is unclear. Consider a caller calling this function:

```cc
auto x = LoadCredsFromPath(path, false, scopes, subjet);
```
It's unclear what the `false` argument means. It probably means "don't *something*", but don't what?

Callers might notice and try to fix this ambiguity by adding a comment to the argument
```cc
auto x = LoadCredsFromPath(path, /* require service accounts */false, scopes, subjet);
```
But this comment doesn't really help because now it's unclear if false means that we should require service accounts, or if we're saying *do not* require service accounts.

All of these problems can usually be fixed using an enum instead of bool. Consider this callsite instead:

```cc
auto x = LoadCredsFromPath(path, AccountType::kServiceAccount, scopes, subject);
```

Now no comment is needed because the code is perfectly clear on its own.

There's more on this topic at https://abseil.io/tips/94